tools/build_config/Jenkinsfile

```diff
@@ -0,0 +1,99 @@
+#!groovy
+
+def post_github_status(String state, String message) {
+    withCredentials([string(credentialsId: 'GitHub_API_Token',
+            variable: 'api_token')]) {
+        sh """
+            curl -H "Authorization: token ${api_token}" \
+                --request POST \
+                --data '{"state": "${state}", \
+                         "description": "${message}", \
+                         "target_url": "$BUILD_URL", \
+                         "context": "$JOB_BASE_NAME"}' \
+                $PR_STATUSES_URL > /dev/null
+        """
+            }
+}
+
+pipeline {
+    agent {
+        label env.AGENT
+    }
+
+    stages {
+
+        stage('Notify') {
+            steps {
+                post_github_status("pending", "The build is running")
+            }
+        }
+
+        stage('Build') {
+            steps {
+                sh '''
+                    module load cmake/\$CMAKE_VERSION &&
+                    module load matlab/\$MATLAB_VERSION &&
+                    module load gcc/\$GCC_VERSION &&
+                    ./tools/build_config/build.sh --build
+                '''
+            }
+        }
+
+        stage('Test') {
+            steps {
+                sh '''
+                    module load cmake/\$CMAKE_VERSION &&
+                    module load matlab/\$MATLAB_VERSION &&
+                    module load gcc/\$GCC_VERSION &&
+                    ./tools/build_config/build.sh --test
+                '''
+            }
+        }
+
+        stage('Package') {
+            steps {
+                sh './tools/build_config/build.sh --package'
+            }
+        }
+    }
+
+    post {
+        always {
```

Comment:
does always happen even if the build fails? does that archive no files.

xunit() is instructing the XUnit plug in to display the archived XML file(s)?
why `deleteOutputFiles: true,` if you're separately archiving them?